### ISSUE TITLE
About self-built dataset

### ISSUE THREAD

**Issue by yoho131** [mala-lab/InCTRL#28]
First of all, thank you to the author for such excellent work!
When I tried to run your code on my self-built dataset, it was missing the Few-shot Normal Samples and Pre-train Models.
So, is there a way to make them?

**Comment by Diana1026** [comment]
Hi, thanks for your attention to this work. If you want to evaluate our model on your self-built dataset, you should first split the training and test datasets by following the form of MVTecAD, where the training set only contains the normal samples and the test set includes both normal and abnormal samples.

Then using our [gen_val_json.py](https://github.com/mala-lab/InCTRL/blob/main/datasets/preprocess/gen_val_json.py) to generate JSON files and the few-shot normal samples are randomly extracted from the training set.

**Comment by yoho131** [comment]
I have run gen_train_json.py and gen_val_json.py on my custom dataset to obtain the four JSON files shown in Figure 1. However, how can I obtain the Few-shot Normal Samples and Pre-train Models required in Step 3 and Step 4?
![image](https://github.com/user-attachments/assets/34e65a1c-e2bf-4d7f-948b-7d37896a88e8)
![image](https://github.com/user-attachments/assets/0e2352b6-d24e-4997-8e14-7b0d00c22cdc)

**Comment by yoho131** [comment]
Sorry, I misunderstood. Steps 1-5 are for testing.

**Comment by yoho131** [comment]
Hello, after running main.py, I obtained the following ten .pyth files. However, when I tried to execute test.py on my custom dataset, I found that the Few-shot Normal Samples mentioned in Step 3 are still missing (.pt files).
![image](https://github.com/user-attachments/assets/6a6dd60f-c052-48ee-a119-3ab47e244fd7)

**Comment by Diana1026** [comment]
Hi, you can write a script to randomly select the few-shot samples from training set and save the image features after your own image transformation. The dimensions of image features should be aligned with the input of the visual encoder.
